### PR TITLE
SW-3895-table Add plantings progress table

### PR DESCRIPTION
**Pull request by karthikbtf**
- added table
- table selection etc is a follow up
- renderer with link to withdrawals (this requires more work on withdrawals component as a follow up)
- cleaned up some selectors
- linked with search

Terraware App 2023-07-18 16-11-27

Terraware App 2023-07-18 16-10-59

Terraware App 2023-07-18 16-10-27

Terraware App 2023-07-18 16-08-28

### CODE REVIEW

File: src/components/NurseryWithdrawals/PlantingProgressList.tsx [hunk 1]
```diff
@@ -1,7 +1,116 @@
...
+
+  const hasZones = data?.some((d) => d.subzoneName);
+
+  if (!data) {
+    return <CircularProgress sx={{ margin: 'auto' }} />;
+  }
+
+  return (
+    <Box>
+      <Table
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/PlantingProgressList.tsx:87]: selected rows changes can be done later, with buttons to complete planting etc

File: src/components/NurseryWithdrawals/PlantingProgressList.tsx [hunk 2]
```diff
@@ -1,7 +1,116 @@
...
+const DetailsRenderer =
+  (classes: any) =>
+  (props: RendererProps<TableRowType>): JSX.Element => {
+    const { column, row } = props;
+
+    const createLinkToWithdrawals = () => {
+      const filterParam = row.subzoneName
+        ? `subzoneName=${encodeURIComponent(row.subzoneName)}`
+        : `siteName=${encodeURIComponent(row.siteName)}`;
+      const url = `${APP_PATHS.NURSERY_WITHDRAWALS}?tab=${strings.WITHDRAWAL_HISTORY}&${filterParam}`;
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/PlantingProgressList.tsx:107]: the filter param is not used here, will need a follow up PR in withdrawals to filter by param and also clear the param if not relevant

File: src/types/PlantingSite.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ export type PlantingProgressSubzone = {
   subzoneName: string;
   plantingCompleted: boolean;
   plantingSite: string;
-  zone: string;
+  zoneName: string;
```
Comment (karthikbtf) [inline src/types/PlantingSite.ts:27]: for consistency

File: src/components/NurseryWithdrawals/PlantingProgressList.tsx (hunk 1 above)
Comment (constanzauanini) [reply src/components/NurseryWithdrawals/PlantingProgressList.tsx:87]: Yes, there is a different ticket for that

**Review by constanzauanini — APPROVED** [review]
Table looks good!

**Comment by karthikbtf** [comment]
> Table looks good!

Thanks @constanzauanini - merging this in so you can rebase and test with your selectors.
Lmk if there's are any conflicts needing my assistance.

**Pull request merged**